src/types.ts

```diff
@@ -1531,6 +1532,7 @@ export type ChannelData<ChannelType = UnknownType> = ChannelType & {
 
 export type ChannelMembership<UserType = UnknownType> = {
   banned?: boolean;
+  channel_role?: string;
```

Comment:
should we extend this to the default role we have? sth like:
```suggestion
  channel_role?: string | 'admin' | 'moderator' | ......;
```

Reply:
We are prepearing infrastructure for new permissions implementation which includes custom roles. These you cannot describe with a enum I'm afraid. We will still have builtin roles though. Can we  express something like this as type?

Reply:
yea you can have a union of builtin roles with a string type